src/main.rs

```diff
@@ -533,77 +633,20 @@ fn main() -> ! {
     spi_drv.init();
     spi_drv.reset(&mut delay);
 
-    // Turn the ESP32's onboard multi-color LED off
-    set_led(&mut spi_drv, &mut uart, 0, 0, 0);
-    delay.delay_ms(500);
-
-    set_led(&mut spi_drv, &mut uart, 255, 0, 0);
-    delay.delay_ms(1000);
-    set_led(&mut spi_drv, &mut uart, 0, 255, 0);
-    delay.delay_ms(1000);
-    set_led(&mut spi_drv, &mut uart, 0, 0, 255);
-    delay.delay_ms(1000);
-    set_led(&mut spi_drv, &mut uart, 100, 100, 100);
-
-    uart.write_full_blocking(b"-----------------\r\n");
-    writeln!(uart, "START_CMD 0x{:X?}\r", START_CMD).ok().unwrap();
-    writeln!(uart, "END_CMD 0x{:X?}\r", END_CMD).ok().unwrap();
-    writeln!(uart, "REPLY_FLAG 0x{:X?}\r", REPLY_FLAG).ok().unwrap();
-    uart.write_full_blocking(b"-----------------\r\n");
-
-    // --- get_fw_version() ---
-    uart.write_full_blocking(b"wait_for_esp_select()\r\n");
-    spi_drv.wait_for_esp_select();
-    uart.write_full_blocking(b"\tesp selected\r\n");
-
-    uart.write_full_blocking(b"send_cmd(GET_FW_VERSION)\r\n");
-    let results = spi_drv.send_cmd(&mut uart, GET_FW_VERSION, 0);
-    match results {
-        Ok(_) => { uart.write_full_blocking(b"\tsent GET_FW_VERSION command\r\n"); }
-        Err(e) => { writeln!(uart, "\t** Failed to send GET_FW_VERSION command: {:?}\r\n", e).ok().unwrap(); }
```

Comment:
Keep these lines as well, but maybe consider moving them into a `get_fw_version()` function like we have for `wifi_set_passphrase` and then call it from here to get the firmware version. Same applies for the other lines below that are associated with `get_fw_version()`.

Let me know if this suggestion isn't completely clear in what I'm interested in seeing.

Reply:
Resolved in [following commit](https://github.com/Jim-Hodapp-Coaching/esp32-pico-wifi/pull/4/commits/7c32670a93d2e8e3abea238966a55b95df787828)